src/checkers/inference/VariableAnnotator.java

```diff
@@ -166,8 +166,7 @@ public VariableAnnotator(final InferenceAnnotatedTypeFactory typeFactory,
                                                                    varAnnot, this);
 
         this.impliedTypeAnnotator = new ImpliedTypeAnnotator(inferenceTypeFactory, slotManager, existentialInserter);
-        this.constantToVariableAnnotator = new ConstantToVariableAnnotator(realTop, varAnnot, this,
-                                                                           slotManager);
+        this.constantToVariableAnnotator = new ConstantToVariableAnnotator(realTop, varAnnot);
```

Comment:
I've checked the reference of this `getNewConstantToVariableAnnotator ` method. The only usage of this method is within tree annotators (`ByteCodeAnnotator` and `InferenceTreeAnnotator`).  In some cases these tree annotators has the logic of delegating to `realTypeFactory` to compute a real qualifier on the given type, and then get a `ConstantToVariableAnnotator` from `InferenceATF` to replace the real qualifier by equivalent VarAnno.

The point of having this method is because these tree annotators doesn't keep a direct reference of `realTop` and `varAnnot` (which are required by constructing a `ConstantToVariableAnnotator`), so here `InferenceATF` provide this wrapper method for tree annotators easily have a `ConstantToVariableAnnotator`.

However, this is no point of creating a new instance of `ConstantToVariableAnnotator` each time when tree annotators need it, as there is no potential conflicts of using the same instance of `ConstantToVariableAnnotator`.

I changed the wrapper method `InferenceATF#getNewConstantToVariableAnnotator` to `getConstantToVariableAnnotator`, and let `InferenceATF` keep a reference to a `ConstantToVariableAnnotator`, so that we save the constructions.